substrate/pkg/controller/substrate/infrastructure/subnets.go

```diff
@@ -97,11 +102,20 @@ func (s *Subnets) ensureSubnet(ctx context.Context, substrate *v1alpha1.Substrat
 		return describeSubnetsOutput.Subnets[0], nil
 	}
 	// tag required by ELB controller to discover these subnets to configure ELB
+	// TODO: add the public and private ELB tags to public or private subnets separately
```

Comment:
`subnetSpec.Public` can give us if the subnet is public and we can tag subnets accordingly. I would suggest lets do that as part of this PR.

Reply:
Yes, but right now deveks only creates public subnets, so `subnetSpec.Public` will only return `true`. We want to add the `kubernetes.io/role/internal-elb` to the same public subnets just for LBC to discover the subnets during test.